From bbe072ae38a25f3ac44d0c5cc0816126a259f69a Mon Sep 17 00:00:00 2001
From: arun <arun.radhakrishnan@intel.com>
Date: Fri, 11 Mar 2016 12:20:44 -0600
Subject: [PATCH 1/2] iio: accel: kxcjk-1013:Initialize the IEL register on
 driver probe

it was observed that on certain flash-reboot cycles, the device
randomly initialized the value of the IEL register bit within the
INT_CTRL_REG1 causing random failure,
This bit is used to toggle between pulse mode(1) and edge mode(0)
initializing this bit within the driver probe function to fix the
failure.

Change-Id: I015d86cd81f09ac1569b00234abea7d7e9cf2308
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-22404
Signed-off-by: Arun Radhakrishnan <arun.radhakrishnan@intel.com>
Reviewed-on: https://android.intel.com:443/480859
---
 drivers/iio/accel/kxcjk-1013.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/iio/accel/kxcjk-1013.c b/drivers/iio/accel/kxcjk-1013.c
index 50b9102..0c1b7c6 100644
--- a/drivers/iio/accel/kxcjk-1013.c
+++ b/drivers/iio/accel/kxcjk-1013.c
@@ -70,6 +70,7 @@
 #define KXCJK1013_REG_CTRL1_BIT_WUFE	BIT(1)
 #define KXCJK1013_REG_INT_REG1_BIT_IEA	BIT(4)
 #define KXCJK1013_REG_INT_REG1_BIT_IEN	BIT(5)
+#define KXCJK1013_REG_INT_REG1_BIT_IEL	BIT(3)
 
 #define KXCJK1013_DATA_MASK_12_BIT	0x0FFF
 #define KXCJK1013_MAX_STARTUP_TIME_US	100000
@@ -365,7 +366,7 @@ static int kxcjk1013_chip_init(struct kxcjk1013_data *data)
 		ret &= ~KXCJK1013_REG_INT_REG1_BIT_IEA;
 
 	ret = i2c_smbus_write_byte_data(data->client, KXCJK1013_REG_INT_CTRL1,
-					ret);
+					ret | KXCJK1013_REG_INT_REG1_BIT_IEL);
 	if (ret < 0) {
 		dev_err(&data->client->dev, "Error writing reg_int_ctrl1\n");
 		return ret;
-- 
1.9.1

